RTClib.cpp

```diff
@@ -367,6 +367,40 @@ DateTime::DateTime(const __FlashStringHelper *date,
   ss = conv2d(buff + 6);
 }
 
+/**************************************************************************/
+/*!
+    @brief  Constructor for creating a DateTime from an ISO8601 date string.
+
+    This constructor expects its parameters to be a string in the
+    https://en.wikipedia.org/wiki/ISO_8601 format, e.g:
+
+    "2020-06-25T15:29:37"
+
+    Usage:
+
+    ```
+    DateTime dt("2020-06-25T15:29:37");
+    ```
+
+    @note The year must be > 2000, as only the yOff is considered.
+
+    @param iso8601dateTime
+           A dateTime string in iso8601 format,
+           e.g. "2020-06-25T15:29:37".
+
+*/
+/**************************************************************************/
+DateTime::DateTime(const char *iso8601dateTime) {
+  char ref[] = "2000-01-01T00:00:00";
+  memcpy(ref, iso8601dateTime, strlen(ref));
```

Comment:
The third argument should be `min(strlen(ref), strlen(iso8601dateTime))`. Otherwise, if the provided string is too short, `memcpy()` reads past its end, which is undefined behavior.

I tested with `Serial.println(DateTime("2020-06-29").timestamp());` and it printed `2020-06-29T52:00:253`.

Reply:
You are right, i overlooked that. Do you open a pull request?

Reply:
OK, will do.

Reply:
Merci.